.travis.yml

```diff
@@ -1,14 +1,22 @@
-# http://docs.travis-ci.com/user/workers/container-based-infrastructure/
-sudo: false
-
 language: php
 
 matrix:
+  fast_finish: true
   include:
-    - env: DB=mysql; MW=master; SMW=2.5.2
-      php: 7
-    - env: DB=mysql; MW=REL1_27;
+    - env: DB=mysql; MW=master; SMW=~3.1@dev
+      php: 7.3
+    - env: DB=mysql; MW=REL1_31; SMW=~3.0@dev
+      php: 7.0
+    - env: DB=mysql; MW=REL1_27; SMW=~3.0@dev
       php: 5.6
+    - env: DB=mysql; MW=REL1_28; SMW=~3.0@dev; TYPE=coverage
+      php: 7.0
+    - env: DB=sqlite; MW=REL1_29; SMW=~3.0@dev
+      php: 7.1
+    - env: DB=postgres; MW=REL1_30; SMW=~3.0@dev
+      php: 7.2      
+  allow_failures:
+    - env: DB=mysql; MW=master; SMW=~3.1@dev
 
```

Comment:
@mwjames Will it make sense to add PHPUnit to these lines like e.g. in SMT? For a lot of extensions we do not add e.g. `SMW=~3.1@dev`. I am unsure when to use it and when not. I guess this is to be able to specify a distinct branch and if we do not master of SMW is used? I this case SMW could probably be dropped from the lines

Reply:
PHPUnit depends on the require PHP version. MW 1.31 comes with a required PHPUnit version which should work with PHP 7.0 but in some cases MW lacks behind and PHP 7.2/7.3 requires a more recent version of PHPUnit.

So, it all depends, and in some cases you get in discussions with WMF about it [0].

[0] https://phabricator.wikimedia.org/T204420

Reply:
Indeed, I remember this one. I just mixed in permissible PHPUnits. Still not sure about the SMW part ...